controller/lib/hal/hal_stm32.cpp

```diff
@@ -222,15 +219,12 @@ static void InitGPIO() {
 
 // Set or clear the specified digital output
 void HalApi::digitalWrite(BinaryPin pin, VoltageLevel value) {
-  GPIO_Regs *base;
-  int bit;
-
-  switch (pin) {
-  case BinaryPin::SOLENOID:
-    base = GPIO_A_BASE;
-    bit = 11;
-    break;
-  }
+  auto [base, bit] = [&]() -> std::pair<GPIO_Regs *, int> {
+    switch (pin) {
+    case BinaryPin::SOLENOID:
+      return {GPIO_A_BASE, 11};
+    }
+  }();
```

Comment:
That certainly looks much cleaner.  It also doesn't look like C++ to me, but as you can probably tell from my code I tend to mostly use C for my projects.  I'm going to research this syntax a bit to get a better understanding of what's going on here, but I'm pretty confident that you know what you're doing.

Reply:
Some pointers -- this is a lambda that I'm using as an IIFE -- creating and immediately invoking it.  I'm then unpacking the pair using a C++17 structured binding declaration, https://en.cppreference.com/w/cpp/language/structured_binding